packages/sdk/src/transactions/AccountTransactionV0.ts

```diff
@@ -0,0 +1,335 @@
+    return {
+        sender: header.sender.toJSON(),
+        nonce: header.nonce.toJSON(),
+        expiry: header.expiry.toJSON(),
+        energyAmount: header.energyAmount.value,
+        payloadSize: header.payloadSize,
+    };
+}
+
+function headerFromJSON(json: HeaderJSON): Header {
+    return {
+        sender: AccountAddress.fromBase58(json.sender),
+        nonce: SequenceNumber.fromJSON(json.nonce),
+        expiry: TransactionExpiry.fromJSON(json.expiry),
+        energyAmount: Energy.create(json.energyAmount),
+        payloadSize: json.payloadSize,
+    };
+}
+
+/**
+ * Signature type for account transactions.
+ */
+export type Signature = AccountTransactionSignature;
+
+/**
+ * A complete version 0 account transaction with header, type, payload, and signature.
+ */
+type AccountTransactionV0 = {
+    /**
+     * Transaction version discriminant;
+     */
+    readonly version: 0;
+    /**
+     * The transaction header containing metadata for the transaction
+     */
+    readonly header: Header;
+    /**
+     * The transaction payload.
+     */
+    readonly payload: Payload.Type;
+    /**
+     * The signature by the transaction sender on the transaction
+     */
+    readonly signature: Signature;
+};
+
+/**
+ * A complete version 0 account transaction with header, type, payload, and signature.
+ */
+export type Type = AccountTransactionV0;
+
+export type JSON = {
+    readonly version: 0;
+    readonly header: HeaderJSON;
+    readonly payload: Payload.JSON;
+    readonly signature: Signature;
+};
+
+export type UnsignedJSON = {
+    readonly version: 0;
+    readonly header: HeaderJSON;
+    readonly payload: Payload.JSON;
+};
+
+/**
+ * Converts a version 0 account transaction to its intermediary JSON serializable representation.
+ *
+ * @param transaction the transaction to convert
+ * @returns the JSON representation of the transaction
+ */
+export function toJSON(transaction: AccountTransactionV0): JSON;
+export function toJSON(transaction: Unsigned): UnsignedJSON;
+export function toJSON(transaction: AccountTransactionV0 | Unsigned): JSON | UnsignedJSON;
+
+export function toJSON(transaction: AccountTransactionV0 | Unsigned): JSON | UnsignedJSON {
+    const base = {
+        version: 0 as const,
+        header: headerToJSON(transaction.header),
+        payload: Payload.toJSON(transaction.payload),
+    };
+    if (!('signature' in transaction)) {
+        return base;
+    }
+    return { ...base, signature: transaction.signature };
+}
+
+/**
+ * Converts a intermediary JSON serializable representation created with {@linkcode toJSON} back to a
+ * version 0 account transaction.
+ *
+ * @param json the JSON to convert
+ * @param [as] the transaction variant to parse. Defaults to parsing {@linkcode AccountTransactionV0}.
+ * @returns the transaction
+ */
+export function fromJSON(json: JSON, as?: 'signed'): AccountTransactionV0;
+export function fromJSON(json: UnsignedJSON, as: 'unsigned'): Unsigned;
```

Comment:
This seems a bit off to me, overloading the parameters and return type like this...
Does it have to be one function or could it be split into two separate functions?
Similar comment to the other `toJSON`/`fromJSON` variants